tools/cli/domainCommands.go

```diff
@@ -443,16 +443,26 @@ func (d *domainCLIImpl) DescribeDomain(c *cli.Context) {
 }
 
 func (d *domainCLIImpl) ListDomains(c *cli.Context) {
-	domains := d.getAllDomains(c)
 	pageSize := c.Int(FlagPageSize)
 	printAll := c.Bool(FlagAll)
+	printDeprecated := c.Bool(FlagDeprecated)
 	printFull := c.Bool(FlagPrintFullyDetail)
 
+	if printAll && printDeprecated {
+		ErrorAndExit("Cannot specify --all and --deprecated flags at the same time.", nil)
```

Comment:
nit: use `FlagAll` and `FlagDeprecated` in the error msg instead of hardcoding them?